addons/t9n/models/resource.py

```diff
@@ -1,11 +1,36 @@
-from odoo import fields, models
+from odoo import fields, models, api, _, Command
+from odoo.exceptions import ValidationError
+import polib
+import base64
+
+
+def decode_resource_file(resource_file):
+    try:
+        file_content = base64.b64decode(resource_file).decode("utf-8")
+        po_obj = polib.pofile(file_content)
+    except:
+        raise ValidationError(_("The file should be a .po file"))
+    return po_obj
+
+
+def create_messages(resource_file):
+    messages_to_create = []
+    for entry in resource_file:
+        messages_to_create.append({
+            "body": entry.msgid,
+        })
+    operations = [Command.create(message_data)
+                  for message_data in messages_to_create]
+    return operations
 
 
 class Resource(models.Model):
     _name = "t9n.resource"
     _description = "Resource file"
 
-    name = fields.Char("Resource")
+    file_name = fields.Char("File Name")
```

Comment:
If you don't explicitly provide a label for the field (i.e. if you don't provide a `string` attribute), the default label is the capitalized field name (file_name → File Name). That is, you can omit the `string="File Name"` here, as the label is already what you want it to be by default